node/tasks.py

```diff
@@ -192,29 +200,48 @@ def sign_sync_point(sync_point: dict[str, Any]) -> str:
 def _validate_nonsigners_stake(nonsigners_stake: int):
     """Verify the nonsigners' stake."""
     return (
-        100 * nonsigners_stake / zconfig.TOTAL_STAKE <= 100 - zconfig.THRESHOLD_PERCENT or
-        (
-            100 * nonsigners_stake / zconfig.NODES_LAST_DATA["total_stake"] <= 100 - zconfig.THRESHOLD_PERCENT and
-            int(time.time()) - zconfig.NODES_LAST_DATA["timestamp"] <= zconfig.NODES_INFO_SYNC_BORDER
-        )
+            100 * nonsigners_stake / zconfig.TOTAL_STAKE <= 100 - zconfig.THRESHOLD_PERCENT or
+            (
+                    100 * nonsigners_stake / zconfig.NODES_LAST_DATA[
+                "total_stake"] <= 100 - zconfig.THRESHOLD_PERCENT and
+                    int(time.time()) - zconfig.NODES_LAST_DATA["timestamp"] <= zconfig.NODES_INFO_SYNC_BORDER
+            )
     )
 
+
+def get_aggregated_public_key(nodes_info, non_signers: List[str]) -> attestation.G2Point:
+    aggregated_public_key: attestation.G2Point = attestation.new_zero_g2_point()
+    non_signer_ids_set = set(non_signers)
+    for node_id, node in nodes_info.items():
+        if node_id not in non_signer_ids_set:
+            aggregated_public_key = aggregated_public_key + node["public_key_g2"]
+    return aggregated_public_key
+
+
 def is_sync_point_signature_verified(
-    app_name: str,
-    state: str,
-    index: int,
-    batch_hash: str,
-    chaining_hash: str,
-    signature_hex: str,
-    nonsigners: list[str],
+        app_name: str,
+        state: str,
+        index: int,
+        batch_hash: str,
+        chaining_hash: str,
+        tag: int,
+        signature_hex: str,
+        nonsigners: list[str],
 ) -> bool:
+    '''
+    Should first load the state of network nodes info using signature tag
+    '''
+
+    nodes_info = zconfig.get_network_info(tag=tag)
```

Comment:
if a tag is loaded before, it should not be queried from the network and loaded again
there should be some caching mechanism